### PR TITLE
WDFN-38 - HUC pages

### PR DESCRIPTION
**Pull request by danielnaab**
- Create manage.py entrypoint for Flask custom commands
- Move lookup generation into "./manage.py generate_lookups --all"
- Create /hydrological-unit view

### CODE REVIEW

**Comment by danielnaab** [comment]
I'm done now - ready for review.

File: waterdata/tests/test_views.py [hunk 1]
```diff
@@ -129,12 +83,101 @@ def test_5xx_from_water_services(self, r_mock):
...
-        r_mock.assert_called_with(self.test_hostname,
+        r_mock.assert_called_with(app.config['SERVICE_ROOT'],
                                   path='/nwis/site/',
                                   params={'site': self.test_site_number,
                                           'agencyCd': 'USGS',
                                           'siteOutput': 'expanded',
                                           'format': 'rdb'
-                                          }
-                                  )
+                                         }
```
Comment (ayan-usgs) [inline waterdata/tests/test_views.py]: As was mentioned in https://github.com/usgs/waterdataui/pull/40, this was PEP8 compliant before...

File: waterdata/tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+"""
```
[the remark this answers is not in the document]
Comment (danielnaab) [reply waterdata/tests/conftest.py]: It's used directly by `test_huc_lookupgs.test_huc_import` (the app parameter), but also indirectly by fixtures exported by pytest-flask (like the client fixture used by TestHydrologicalUnitView)

File: waterdata/templates/hydrological_unit.html [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                        <tr>
+                            <th scope="col">HUC</th>
+                            <th scope="col">Name</th>
+                        </tr>
+                    </thead>
+                    <tbody>
+                        {% for huc_cd in huc.children %}
+                            <tr>
+                                <th scope="row"><a href="{{ url_for('hydrological_unit', huc_cd=huc_cd) }}">{{ huc_cd }}</a></th>
+                                <td> {{ config.HUC_LOOKUP.hucs[huc_cd].huc_nm }} </td>
```
[the remark this answers is not in the document]
Comment (danielnaab) [reply waterdata/templates/hydrological_unit.html]: In this case, `huc_cd` is a variable.

File: waterdata/utils.py [hunk 4]
```diff
@@ -45,11 +45,14 @@ def parse_rdb(rdb_iter_lines):
         except StopIteration:
             raise Exception('RDB column headers not found.')
         else:
-            if line[0] != '#':
+            if line and line[0] != '#':
```
[the remark this answers is not in the document]
Comment (danielnaab) [reply waterdata/utils.py]: This fixes an issue where the HUC help service returns with a newline at the end of the document.

File: waterdata/tests/conftest.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mbucknell) [reply waterdata/tests/conftest.py]: I'm a pytest newbie. So pytest-flask gets imported as part of the pytest test runner? It seems a little magical at the moment.

[the remark this answers is not in the document]
Comment (danielnaab) [reply waterdata/tests/conftest.py]: Yes, it is a bit. pytest looks for packages prefixed with "pytest-" and manages fixtures they export. When the test runner executes, if it sees a test function that takes a parameter matching a fixture name, it passed it in when running the function.

File: waterdata/tests/commands/lookup_generation/test_commands.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
+"""
```
Comment (danielnaab) [inline waterdata/tests/commands/lookup_generation/test_commands.py]: I added these tests to increase coverage over the lookup generation commands. They just confirm that JSON files are generated when run with some mock RDB responses (defined in waterdata/tests/commands/lookup_generation/conftest.py).

File: config.py [hunk 6]
```diff
@@ -20,3 +20,4 @@
 
 NWIS_CODE_LOOKUP_FILENAME = os.path.join('data', 'nwis_lookup.json')
 COUNTRY_STATE_COUNTY_LOOKUP_FILENAME = os.path.join('data', 'nwis_country_state_lookup.json')
+HUC_LOOKUP_FILENAME = os.path.join('data', 'huc_lookup.json')
```
[the remark this answers is not in the document]
Comment (mbucknell) [reply config.py]: I'd be ok with that...especially since it seems they will be numerous.

**Pull request merged**